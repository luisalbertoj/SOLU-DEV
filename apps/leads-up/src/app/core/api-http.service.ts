import { Injectable } from '@angular/core';
// Atento a esta línea, deberás hacer que empate con tu environment
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { environment } from '../../environments/environment';

import { Observable } from 'rxjs';

const API_URL = environment.apiRoute;

@Injectable({
  providedIn: 'root',
})
export class ApiHttpService {
  protected headers;

  protected apiUrl = API_URL;

  constructor(private http: HttpClient) {
    this.headers = {
      headers: new HttpHeaders({
        'Content-Type': 'application/json',
        Accept: 'application/json',
      }),
    };
  }

  protected get(url: string): Observable<unknown> {
    return this.http.get(this.apiUrl + url, this.headers);
  }

  protected post({ url, payload }: { url: string; payload: unknown }) {
    return this.http.post('api/' + url, payload, this.headers);
  }
}
