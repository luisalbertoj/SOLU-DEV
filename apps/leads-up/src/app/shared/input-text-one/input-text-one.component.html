<span class="p-float-label">
  <input
    id="float-input-invalid"
    type="text"
    pInputText
    [(ngModel)]="model"
    (ngModelChange)="modelChangeData($event)"
    class="ng-invalid ng-dirty"
  />
  <label for="float-input-invalid">Username</label>
</span>
