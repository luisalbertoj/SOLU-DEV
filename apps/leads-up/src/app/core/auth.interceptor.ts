import {
  HttpEvent,
  HttpHandler,
  HttpInterceptor,
  HttpRequest,
} from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable()
export class AuthInterceptor implements HttpInterceptor {
  intercept(
    req: HttpRequest<unknown>,
    next: HttpHandler
  ): Observable<HttpEvent<unknown>> {
    const token = localStorage.getItem('auth_token');

    if (!token) {
      return next.handle(req);
    }

    const request = req.clone({
      headers: req.headers.set('Authorization', `Bearer ${token}`),
    });

    return next.handle(request);
  }
}
