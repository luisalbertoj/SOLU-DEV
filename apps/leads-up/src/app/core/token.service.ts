import { Injectable } from '@angular/core';

@Injectable()
export class TokenProvider {
  load() {
    return new Promise((resolve) => {
      resolve(true);
    });
  }
}
