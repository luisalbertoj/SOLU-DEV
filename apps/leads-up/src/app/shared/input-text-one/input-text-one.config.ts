import { Directive, EventEmitter, Input, Output } from '@angular/core';

@Directive()
export class InputTextOneConfig {
  @Input() model!: string;
  @Output() modelChange = new EventEmitter();

  modelChangeData(data: string) {
    this.modelChange.emit(data);
  }
}
