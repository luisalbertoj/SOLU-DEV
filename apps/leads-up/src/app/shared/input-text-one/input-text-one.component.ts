import { CommonModule } from '@angular/common';
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { InputTextModule } from 'primeng/inputtext';
import { InputTextOneConfig } from './input-text-one.config';

@Component({
  selector: 'solu-dev-input-text-one',
  standalone: true,
  imports: [CommonModule, InputTextModule, FormsModule],
  templateUrl: './input-text-one.component.html',
  styleUrls: ['./input-text-one.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class InputTextOneComponent extends InputTextOneConfig {}
