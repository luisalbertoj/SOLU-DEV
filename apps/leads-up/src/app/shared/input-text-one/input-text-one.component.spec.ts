import { ComponentFixture, TestBed } from '@angular/core/testing';

import { InputTextOneComponent } from './input-text-one.component';

describe('InputTextOneComponent', () => {
  let component: InputTextOneComponent;
  let fixture: ComponentFixture<InputTextOneComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [InputTextOneComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(InputTextOneComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
