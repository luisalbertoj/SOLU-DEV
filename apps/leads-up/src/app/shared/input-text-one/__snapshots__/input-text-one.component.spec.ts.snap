// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTextOneComponent this should be equal to the snapshot 1`] = `"<input pinputtext=\\"\\" class=\\"p-inputtext p-component p-element ng-untouched ng-pristine ng-valid\\" type=\\"undefined\\" autocomplete=\\"undefined\\" placeholder=\\"undefined\\" ng-reflect-ng-class=\\"ng-dirty ng-invalid\\">"`;
